sdk/keyvault/Azure.Security.KeyVault.Secrets/src/SecretClient.cs

```diff
@@ -258,16 +329,49 @@ private async Task<Response> SendRequestAsync(Request request, CancellationToken
                     throw await response.CreateRequestFailedExceptionAsync();
             }
         }
+        private Response SendRequest(Request request, CancellationToken cancellationToken)
+        {
+            var response = _pipeline.SendRequest(request, cancellationToken);
 
+            switch (response.Status)
+            {
+                case 200:
+                case 201:
+                    return response;
+                default:
+                    throw response.CreateRequestFailedException();
+            }
+        }
 
-        private async Task<Response<Page<T>>> GetPageAsync<T>(Uri pageUri, Func<T> itemFactory, CancellationToken cancellationToken)
-            where T : Model
+        private async Task<PageResponse<T>> GetPageAsync<T>(Uri pageUri, Func<T> itemFactory, CancellationToken cancellationToken)
+                where T : Model
         {
             using (Request request = CreateRequest(HttpPipelineMethod.Get, pageUri))
             {
                 Response response = await SendRequestAsync(request, cancellationToken);
 
-                return this.CreateResponse(response, new Page<T>(itemFactory));
+                // read the respose
```

Comment:
We should check the status of the response too. See as example: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/appconfiguration/Azure.ApplicationModel.Configuration/src/ConfigurationClient.cs#L517

Reply:
Done. There is code in `SendRequestAsync` that checks the status before returning the response.